### ISSUE TITLE
[HPM] Upgrading to WebSocket error

### ISSUE THREAD

**Issue by maqingbo** [probil/vue-socket.io-extended#427]
Hello author, I am very happy to use socket.io in vue through this package. But when I use this package in vue-cli，I got an error.

```js
// vue.config.js :

module.exports = {
  publicPath: './',
  productionSourceMap: false,
  devServer: {
    port: 5013, // 配置端口
    // 设置让浏览器 overlay 同时显示警告和错误
    overlay: {
      warnings: true,
      errors: true
    },
    proxy: {
      '/socket.io': {
        target: 'http://10.277.219.88:5013',
        changeOrigin: true,
        ws: true,
        logLevel: 'debug'
      }
    }
  }
}
```

```js
// main.js

import VueSocketIOExt from 'vue-socket.io-extended'
import SocketIO from 'socket.io-client'

const socket = SocketIO('', { path: '/socket.io' })
Vue.use(VueSocketIOExt, socket)
```

```bash
// error

  App running at:
  - Local:   http://localhost:5013
  - Network: http://10.240.253.229:5013

  Note that the development build is not optimized.
  To create a production build, run npm run build.

[HPM] GET /socket.io/?EIO=3&transport=polling&t=Mvji_cx -> http://10.237.219.77:5013
[HPM] GET /socket.io/?EIO=3&transport=polling&t=Mvjj1pK&sid=4398dbbaf9bf4fcb89a44f53c05331ea -> http://10.237.219.77:5013
[HPM] GET /socket.io/?EIO=3&transport=websocket&sid=4398dbbaf9bf4fcb89a44f53c05331ea -> http://10.237.219.77:5013
[HPM] Upgrading to WebSocket
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:111:27)
Emitted 'error' event at:
    at errorOrDestroy (internal/streams/destroy.js:107:12)
    at Socket.onerror (_stream_readable.js:717:7)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! vue-manage-system@4.0.0 serve: `vue-cli-service serve`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the vue-manage-system@4.0.0 serve script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

When I fill in the IP address directly in main.js, I am prompted to cross-domain issues, so I use proxy, I guess the problem might be in the proxy, but I didn't find a solution. Is my configuration wrong?

**Comment by probil** [comment]
Hi @maqingbo :wave:

I don't recommend using `proxy` because you wouldn't be able to use it in production. Basically are delaying problem solving for later times.

Can you tell me what kind of cross-domain issues do you have? :thinking: 

P.S. You can make it work with `proxy` by passing `upgrade: false` to socket.io constructor but as I've already mentioned don't recommend it.

**Comment by maqingbo** [comment]
My English is not good, it may not be very accurate（cross-domain）.I'll show you the picture：

![2019-11-18 18 08 35](https://user-images.githubusercontent.com/20136852/69044974-df57c280-0a30-11ea-960c-aca3b9e6bdb7.png)

The url I got is `http` not `ws`，so I need to upgrade from long polling to websocket，this problem is generated when sending long polls.

**Comment by probil** [comment]
@maqingbo 
Your server should respond with a special header so it's not blocked by browser. The easiest way to do so is to pass `origins` options to the socket.io constructor on the server:
```js

// replace
var io = require('socket.io')(server);
// with this
var io = require('socket.io')(server, { origins: 'http://your-front-end-domain.com'});
// or with this (less secure)
var io = require('socket.io')(server, { origins: '*:*'}); 
```
https://stackoverflow.com/questions/24058157/socket-io-node-js-cross-origin-request-blocked

**Comment by probil** [comment]
Closing the issue because of inactivity. 
Feel free to create a new issue if you have any other problems.
